### PR TITLE
Split parsing logic into separate method

### PR DESCRIPTION
**Pull request by slick-nick-90**
Allow parsing makemkv that exists already. This change allows me to do that

### CODE REVIEW

**Comment by d-k-bo** [comment]
Thank you for your interest in contributing to this project!

Could you please explain why you think the parsing logic should be split into a separate method? I admit that `MakeMKV._run()` is a huge block of code, but I am not sure if this change is the right approach to solve the problem (the for loop is still ~200 lines long).

File: makemkv/types.py [hunk 1]
```diff
@@ -48,6 +48,7 @@ class Title(TypedDict, total=False):
     information: str
     length: str  # hh:mm:ss
     size_human: str  # eg. "5.9 GB"
+    segment_map: str  # eg. "00021.mpls"
```
[the remark this answers is not in the document]
Comment (slick-nick-90) [reply makemkv/types.py:51]: I wanted to be able to parse a makemkv log I already made. This change allows me to do that

[the remark this answers is not in the document]
Comment (slick-nick-90) [reply makemkv/types.py]: good call, Fixed!

File: makemkv/makemkv.py [hunk 2]
```diff
@@ -246,17 +246,16 @@ def _translate_codes(
 
         return key, return_value
 
-    def _run(self, cmd: list[str]) -> MakeMKVOutput:
-        """Run makemkvcon and parse its output."""
-        p = self.process = Popen(cmd, stderr=STDOUT, stdout=PIPE, bufsize=1, text=True)
-        logger.info('Running "%s"', " ".join(cmd))
+    def _parse_makemkv(self, lines, p=None) -> dict:
```
Comment (d-k-bo) [inline makemkv/makemkv.py:249]: `_parse_makemkv_log` would be a more precise name.

Could you please provide type annotations? I think `lines` should be `Iterable[str]` (with `Iterable` being imported from `typing`). Return type should be `MakeMKVOutput`.

`p` can be ommited and all references should be replaced with `self.process`.

So the overall method signature would look like
```python
def _parse_makemkv_log(self, lines: Iterable[str]) -> MakeMKVOutput:
```

File: makemkv/makemkv.py [hunk 3]
```diff
@@ -283,7 +282,7 @@ def _run(self, cmd: list[str]) -> MakeMKVOutput:
                 loglevel = MESSAGE_CODES.get(code, 10)
                 makemkvcon_logger.log(loglevel, "%s (%s)", message, code)
 
-                if loglevel == logging.CRITICAL:
+                if loglevel == logging.CRITICAL and p !=None:
```
Comment (d-k-bo) [inline makemkv/makemkv.py:285]: To check if a value is None, `is` / `is not` should be used (see [PEP 8](https://peps.python.org/pep-0008/#programming-recommendations)), so this should be `if loglevel == logging.CRITICAL and self.process is not None:`.

**Review by d-k-bo — CHANGES_REQUESTED** [review]
It would be great if you could also run [`nox`](https://nox.thea.codes/en/stable/tutorial.html#installation) on your code to fix formatting issues and try to fix linter warnings if there are any.

**Review by d-k-bo — APPROVED** [review]

File: makemkv/makemkv.py [hunk 4]
```diff
@@ -283,7 +282,7 @@ def _run(self, cmd: list[str]) -> MakeMKVOutput:
                 loglevel = MESSAGE_CODES.get(code, 10)
                 makemkvcon_logger.log(loglevel, "%s (%s)", message, code)
 
-                if loglevel == logging.CRITICAL:
+                if loglevel == logging.CRITICAL and self.process is not None:
                     p.kill()
```
Comment (d-k-bo) [inline makemkv/makemkv.py:286]: This should be `self.process.kill()`

File: makemkv/makemkv.py [hunk 5]
```diff
@@ -246,17 +246,16 @@ def _translate_codes(
 
         return key, return_value
 
-    def _run(self, cmd: list[str]) -> MakeMKVOutput:
-        """Run makemkvcon and parse its output."""
-        p = self.process = Popen(cmd, stderr=STDOUT, stdout=PIPE, bufsize=1, text=True)
-        logger.info('Running "%s"', " ".join(cmd))
+    def _parse_makemkv_log(self, lines: Iterable[str]) -> MakeMKVOutput:
```
Comment (d-k-bo) [inline makemkv/makemkv.py:249]: `Iterable` needs to be imported in line 12

**Review by d-k-bo — CHANGES_REQUESTED** [review]

**Comment by d-k-bo** [comment]
There are still issues with code formatting. Could you please run `nox -s format` or `black .` on your code?

**Review by d-k-bo — APPROVED** [review]

**Comment by slick-nick-90** [comment]
> It would be great if you could also run [`nox`](https://nox.thea.codes/en/stable/tutorial.html#installation) on your code to fix formatting issues and try to fix linter warnings if there are any.

I ran it and fixed it

**Pull request merged**